src/oracle/OracleWriter.sol

```diff
@@ -17,19 +17,34 @@ abstract contract OracleWriter is Pair, IOracleWriter {
     uint256 internal constant MAX_CHANGE_PER_SEC = 0.01e18;
     string internal constant ALLOWED_CHANGE_NAME = "Shared::allowedChangePerSecond";
 
-    Observation[65536] public observations;
+    Observation[65536] internal _observations;
     uint16 public index = type(uint16).max;
 
+    mapping(address => bool) authorizedCallers;
```

Comment:
I don't like using a mapping, can we just have a single address called `OracleReader` that is authorized? That address can then be changed to support contract upgrading.

In general we should model our system like MakerDAO does, small components that can be swapped out by simply changing an address.